src/lib.rs

```diff
@@ -18,6 +23,19 @@ pub use beacon::*;
 
 pub type Result<T = ()> = std::result::Result<T, Error>;
 
+pub enum Payload {
+    CellAttach(CellAttach),
+    CellScan(CellScan),
+    Beacon(Beacon),
+}
+
+pub struct Message {
```

Comment:
Could you please `#[derive(debug)]` on some of these structs (I find it useful in general)

Reply:
good point